javascript/ql/src/semmle/javascript/dataflow/internal/InterProceduralTypeInference.qll

```diff
@@ -147,39 +147,15 @@ predicate mayReturnImplicitly(Function f) {
 /**
  * A call with inter-procedural type inference for the return value.
  */
-abstract class CallWithAnalyzedReturnFlow extends DataFlow::CallNode, DataFlow::AnalyzedValueNode {
+abstract class CallWithAnalyzedReturnFlow extends DataFlow::AnalyzedValueNode {
 
   /**
    * Gets a called function.
    */
-  abstract Function getAFunction();
-  
-  /**
-   * Gets a return value for this call.
-   */
-  AbstractValue getAReturnValue() {
-    exists (Function f | f = getAFunction() |
-      if f.isGenerator() or f.isAsync() then
-        result = TAbstractOtherObject()
-      else (
-        // explicit return value
-        result = f.getAReturnedExpr().analyze().getALocalValue()
-        or
-        // implicit return value
-        (
-          // either because execution of the function may terminate normally
-          mayReturnImplicitly(f)
-          or
-          // or because there is a bare `return;` statement
-          exists (ReturnStmt ret | ret = f.getAReturnStmt() | not exists(ret.getExpr()))
-        ) and
-        result = TAbstractUndefined()
-      )
-    )
-  }
+  abstract Function getACallee();
```

Comment:
Why not just make this return an `AnalyzedFunction`?

Reply:
Yeah, it's probably fine to expose that. Although I suppose we should move it out of `BasicExprTypeInference.qll`, which is in `internal`.